core/src/main/java/org/infinispan/interceptors/distribution/TxDistributionInterceptor.java

```diff
@@ -102,6 +104,16 @@ public Object visitReplaceCommand(InvocationContext ctx, ReplaceCommand command)
       return handleTxWriteCommand(ctx, command, command.getKey());
    }
 
+   @Override
+   public Object visitComputeCommand(InvocationContext ctx, ComputeCommand command) throws Throwable {
+      return handleTxWriteCommand(ctx, command, command.getKey());
```

Comment:
For the record: this does not work as in-place invocation on transactional caches; `handleTxWriteCommand` will retrieve the full value to a non-owner node.